lib/api/pokedex_api_spec/handler/api_client.dart

```diff
@@ -0,0 +1,17 @@
+import 'package:dio/dio.dart';
+
+class ApiClient {
+  final String baseUrl;
+  late Dio dio;
+
+  ApiClient(
+    this.baseUrl,
```

Comment:
I think there's no need to have this `baseUrl` as a constructor param, unless you think that this will importantly be coming from an environment variable or secret key injection from github secrets🤔, rather having it in a private static const is enough I would say 🙂